tests/integration/dynamodb2/test_highlevel.py

```diff
@@ -716,3 +716,68 @@ def test_query_after_describe_with_gsi(self):
 
         for rs_item in rs:
             self.assertEqual(rs_item['username'], ['johndoe'])
+
+    def test_update_table_online_indexing_support(self):
+        # Create a table using gsi to test the DynamoDB online indexing support
+        # https://github.com/boto/boto/issues/2828
+        users = Table.create('online_indexing_support_users', schema=[
+            HashKey('user_id')
+        ], throughput={
+            'read': 5,
+            'write': 5
+        }, global_indexes=[
+            GlobalAllIndex('EmailGSIIndex', parts=[
+                HashKey('email')
+            ], throughput={
+                'read': 2,
+                'write': 2
+            })
+        ])
+
+        # Add this function to be called after tearDown()
+        self.addCleanup(users.delete)
+
+        # Wait for it.
+        time.sleep(60)
+
+        # Update a GSI throughput. it should work.
+        users.update_global_secondary_index(global_index={
+            'EmailGSIIndex': {
+                'read': 2,
+                'write': 1,
+            }
+        })
+
+        # Wait for it.
+        time.sleep(60)
+
```

Comment:
For tests, we typically like it if you explicitly assert something as opposed to just testing that you can successfully call the method. I would suggest using the table's layer2 to make a call to ``describe_table()`` and then asserting that the response has the global secondary indexes that you are expecting.